include/oneapi/dpl/pstl/hetero/dpcpp/execution_sycl_defs.h

```diff
@@ -247,6 +253,10 @@ struct __is_device_execution_policy<execution::device_policy<PolicyParams...>> :
 template <typename... PolicyParams>
 inline constexpr bool __is_device_execution_policy_v = __is_device_execution_policy<PolicyParams...>::value;
 
+template <typename _ExecPolicy, const bool __condition, typename _T = void>
+using __enable_if_device_execution_policy_conditional = ::std::enable_if_t<
+    oneapi::dpl::__internal::__is_device_execution_policy_v<::std::decay_t<_ExecPolicy>> && __condition, _T>;
```

Comment:
Is there a reason this is inserted as an extra layer and not added in to `__is_device_execution_policy_v` to just decay the incoming `PolicyParams` there?
  
Or probably better still, make an "impl" layer of  `__is_device_execution_policy` to do the actual template specialization, and add the decay to  `__is_device_execution_policy`.

It doesn't make sense to me why this `conditional` version gets the automatic decay, but the others don't.  I'm not sure the `conditional` version needs to exist even though it seems to be used in concert with other conditions.

Also, on the topic of `__is_device_execution_policy_v` ... shouldn't that template parameter be named `_ExecPolicy` , and not be a variadic template?  (This is not part of the PR, but lets fix it if its broken.)

Reply:
Fixed

Reply:
This is better, but I think we can make it slightly better still.  Perhaps something like this? (whatever is done should also be done for `__is_hetero_execution_policy`
```
template <typename _T>
struct __is_device_execution_policy_impl : ::std::false_type
{
};

template <typename... PolicyParams>
struct __is_device_execution_policy_impl<execution::device_policy<PolicyParams...>> : ::std::true_type
{
};


template <typename _ExecPolicy>
using __is_device_execution_policy = __is_device_execution_policy_impl<::std::decay_t<_ExecPolicy>>;


template <typename _ExecPolicy>
inline constexpr bool __is_device_execution_policy_v = __is_device_execution_policy<_ExecPolicy>::value;
```


This does 2 things beyond what is currently in the PR:
1) Adds the decay to `__is_device_execution_policy` so that it can be consistent in whatever flavor someone may want
2) Changes the variadic `PolicyParams` template argument to `_ExecPolicy` at higher levels, where I believe we always want exactly one Execution Policy as input.  (this could be its own PR but why not improve things while we are here)

Reply:
Looks like the are not exactly the same:
```C++
struct __is_hetero_execution_policy : ::std::false_type
struct __is_hetero_execution_policy<execution::device_policy<PolicyParams...>> : ::std::true_type
 
struct __is_device_execution_policy : ::std::false_type
struct __is_device_execution_policy<execution::device_policy<PolicyParams...>> : ::std::true_type
struct __is_hetero_execution_policy<execution::fpga_policy<unroll_factor, PolicyParams...>> : ::std::true_type
```

Reply:
Great, I think this looks good.  Thanks.

I've created #1294 to track an issue I brought up here which is beyond the scope of this PR, and could be looked at later.

Reply:
Probably you mean https://github.com/oneapi-src/oneDPL/issues/1298 ?

Reply:
Yes you are correct, I made a typo.  #1298.